modules/iam-role-for-service-accounts-eks/policies.tf

```diff
@@ -531,24 +531,23 @@ data "aws_iam_policy_document" "karpenter_controller" {
       "ec2:DescribeInstanceTypeOfferings",
       "ec2:DescribeAvailabilityZones",
       "ec2:DescribeSpotPriceHistory",
+      "ec2:RunInstances",
+      "iam:PassRole",
```

Comment:
this is already available here https://github.com/terraform-aws-modules/terraform-aws-iam/blob/d6164989a94e1e9711e40e6fb0c374c2d3945149/modules/iam-role-for-service-accounts-eks/policies.tf#L586

Reply:
@bryantbiggs The Karpenter controller IAM role is using the policies defined [here](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/9210e6c6dc2bbd00065bc6f9212d04a0f49adec2/modules/iam-role-for-service-accounts-eks/policies.tf#L517). When using that policy karpenter has lots of unauthorized errors and can not even spin up nodes. However,  we use the policy defined in the [AWS EKS Blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/015834f05273cab73310cec68386aebf7d64627b/modules/kubernetes-addons/karpenter/data.tf) karpenter works well.

The current IRSA module for karpenter does not work out of the box for us because of the missing permissions. I can show you some evidence that the current module `5.3.1` does not have the right permissions in place.

Reply:
can you provide your configuration so I can try to reproduce on my end?